src/cobalt.rs

```diff
@@ -181,40 +201,43 @@ fn generate_pages(posts: Vec<Document>, documents: Vec<Document>, context: &Cont
     trace!("Generating other documents");
     for mut doc in documents {
         trace!("Generating {}", doc.url_path);
-        if doc.front.pagination.is_some() {
-            let paginators = pagination::generate_paginators(&mut doc, &posts_data)?;
-            // page 1 uses frontmatter.permalink instead of paginator.permalink
-            let mut paginators = paginators.into_iter();
-            let paginator = paginators
-                .next()
-                .expect("We detected pagination enabled but we have no paginator");
+        if doc.front.pagination.order == SortOrder::None {
+            // "inherit" sort order from frontmatter if none set in `pagination` section
+            doc.front.pagination.order = context.posts.order;
```

Comment:
Might not be feasible but can we implement this over in `config.rs` where we merge in the posts-defaults?

Reply:
It's not possible in `config.rs` but I can make it happen in `document.rs` by modifying `Document::parse`: instead of `default_front: cobalt_model::FrontmatterBuilder`, I can pass the whole `Collection` so I have both `default_front` extracted from it and the `posts.order` flag to pass to `FrontmatterBuilder` somehow (either a `set_pagination_order` function or a parameter in `build()`).

Shall I do it?

Reply:
Why can't it be done in `config.rs`?

Reply:
I can't find my way while reading the file. As far as I understand it, I need access to `DocumentBuilder` which can't be done in `config.rs`?

I might be  wrong though

Reply:
Shall I proceed on moving it in `document.rs` or do we keep it here?

Reply:
@epage (I mention you, as it begins to be long without answer, maybe you didn't get the notif, like the other time :) )